cl/corpus_importer/management/commands/import_tn.py

```diff
@@ -0,0 +1,188 @@
+                case["neutral_citation"] = "%s %s %s" % (key, reporter, count)
+                results.append(case)
+                count += 1
+    return results
+
+
+def import_tn_corpus(log, skip_until, dir):
+    """Import TN Corpus
+
+    :param log: Should we view logging info
+    :param skip_until: Label ID, if any, to process first
+    :param dir: Location of our import files
+    :return: None
+    """
+    ready = False if skip_until else True
+
+    if log:
+        logging.getLogger().setLevel(logging.INFO)
+
+    logging.info("Starting import")
+    filepath = "%s/data.json" % dir
+    tn_corpus = add_neutral_citations(filepath)
+    if not ready:
+        case = [x for x in tn_corpus if x["label"] == int(skip_until)][0]
+        logging.info(
+            "Skipping until case %s labeled: %s", case["title"], case["label"]
+        )
+
+    for case in sorted(tn_corpus, key=lambda x: x["label"]):
+        if case["label"] == int(skip_until):
+            ready = True
+        if not ready:
+            continue
+        logging.info(
+            "Processing label:%s for case:%s", case["label"], case["title"]
+        )
+        pdf_path = [
+            x
+            for x in glob("%s/%s/*.pdf" % (dir, case["label"]))
+            if "stamped" not in x
+        ][0]
+        pdf_data = open(pdf_path, "r").read()
+
+        sha1_hash = sha1(force_bytes(pdf_data))
+        ops = Opinion.objects.filter(sha1=sha1_hash)
+        if len(ops) > 0:
+            op = ops[0]
+            logging.warn(
+                "Document already in database. See: %s at %s"
+                % (op.get_absolute_url(), op.cluster.case_name)
+            )
+
+        docket, opinion, cluster, citations, error = make_objects(
+            make_item(case),
+            Court.objects.get(pk=case["court"]),
+            sha1(force_bytes(pdf_data)),
+            open(pdf_path, "r").read(),
+        )
+
+        save_everything(
+            items={
+                "docket": docket,
+                "opinion": opinion,
+                "cluster": cluster,
+                "citations": citations,
+            },
+            index=False,
+        )
+
+        extract_doc_content.delay(
+            opinion.pk, do_ocr=True, citation_jitter=True,
+        )
+        logging.info(
+            "Successfully added Tennessee object cluster: %s", cluster.id
+        )
+
+
+class Command(VerboseCommand):
+    help = "Import TN data corpus received from TN Workers Comp boards."
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--input-dir",
+            help="The directory containing our data.",
+            required=True,
+        )
+        parser.add_argument(
+            "--log",
+            action="store_true",
+            default=False,
+            help="Choose to view info log lines.",
+        )
+        parser.add_argument(
+            "--skip-until", default=False, help="Skip until to process",
```

Comment:
Make this an int, and make it default to 0 or -1. Should simplify things above and it'll provide better argparse'ing since non-ints won't be allowed.

Reply:
√